core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/LoggingPlugin.kt

```diff
@@ -16,17 +16,33 @@
 package com.expediagroup.sdk.core.plugin.logging
 
 import com.expediagroup.sdk.core.client.Client
+import com.expediagroup.sdk.core.constant.ExceptionMessage
+import com.expediagroup.sdk.core.constant.LoggingMessage
+import com.expediagroup.sdk.core.model.exception.service.ExpediaGroupAuthException
 import com.expediagroup.sdk.core.plugin.Plugin
 import io.ktor.client.plugins.logging.Logging
 
 internal object LoggingPlugin : Plugin<LoggingConfiguration> {
+    val clientLoggingMaskedFieldsProviders = mutableMapOf<Client, LoggingMaskedFieldsProvider>()
```

Comment:
Why `mutable`?

Reply:
Because we add LoggingMaskedFieldsProvider to it once they're created per client creation